### PR TITLE
[GEP-18] Adapt `etcd-encryption-secret` to new secrets manager

### LINKED ISSUE #1578: Optimize etcd encryption handling
With https://github.com/gardener/gardener/pull/1066 we have introduced etcd encryption for shoot clusters. Today, in order to activate the encryption for new shoot clusters we require two reconciliation flows. In the first flow the encryption configuration is created but set to inactivate, and only in the second flow it's made active. However, it would be possible to combine that into one flow, at least for new shoots (for existing shoots we should keep it as it is).

Also, we should parallelize the secret rewriting in the shoot cluster in case the etcd encryption config has changed. This has already been done with https://github.com/gardener/gardener/pull/1391 but was reverted with 8420df1a0a02151bd14966057eae760d42d52773 due to instabilities. Let's fix them and reintroduce it.

### PR DESCRIPTION
**Pull request by rfranzke**
**How to categorize this PR?**

/area security
/kind enhancement
/merge squash

**What this PR does / why we need it**:
This PR adapts the `etcd-encryption-secret` secrets to new secrets manager. It moves the generation into the `kubeapiserver` component since it actually belongs there (this is the only component that uses it).

The feature to force the `kube-apiserver` of shoot clusters to write secrets in plain text to etcd was dropped since it probably was never used, is undocumented and anyways questionable security-wise. 

Also, the `botanist.RewriteShootSecretsIfEncryptionConfigurationChanged` function was dropped entirely. Back then, it was introduced due to the following scenarios:

1) Old shoots with unencrypted etcd data will now get the encryption configuration (hence, secrets must be rewritten to become encrypted in etcd)
2) Encryption key changes/rotates
3) Secrets were forced to plain text (via `shoot.gardener.cloud/etcd-encryption-force-plaintext-secrets=true` annotation on the `etcd-encryption-secret` in the shoot namespace in the seed)

Now, 1) is no longer valid since we only support shoots with encrypted etcd data since very long. 2) is not implemented. 3) will be dropped (this feature was never used and is not too helpful anyways)

Given that the code is not optimal (sequential instead of parallel) and that it wouldn't work when https://github.com/gardener/gardener/issues/4606 gets implemented, let's drop it now. It has no unit tests anyways, so let's rather invest the time to write it properly when we either implement encryption key rotation or support end-user configuration for additional resources that should be encrypted (https://github.com/gardener/gardener/issues/4606). See also #1578.

**Which issue(s) this PR fixes**:
Part of https://github.com/gardener/gardener/issues/3292#issuecomment-1036058890
Fixes #1578

/invite @timebertt @BeckerMax 
/invite @plkokanov 

**Release note**:

```noteworthy operator
It is no longer possible to force the `kube-apiserver` of shoot clusters to write secrets in plain text to etcd.
```

### CODE REVIEW

File: pkg/utils/secrets/manager/generate.go [hunk 1]
```diff
@@ -214,6 +219,49 @@ func (m *manager) keepExistingSecretsIfNeeded(ctx context.Context, configName st
...
+				if provider.AESCBC != nil && len(provider.AESCBC.Keys) != 0 {
+					existingEncryptionKey = []byte(provider.AESCBC.Keys[0].Name)
+					existingEncryptionSecret = []byte(provider.AESCBC.Keys[0].Secret)
+					break
+				}
+			}
+		}
+
+		if existingEncryptionKey == nil || existingEncryptionSecret == nil {
+			return nil, fmt.Errorf("old etcd encryption key could not found be retrieved")
```
Comment (plkokanov) [inline pkg/utils/secrets/manager/generate.go]:
```suggestion			
			return nil, fmt.Errorf("old etcd encryption key or secret could not be found")
```

**Review by plkokanov — CHANGES_REQUESTED** [review]
I just had a rough pass of the PR. I'll have to get a bit more up-to-date with the new secret manager and do a second pass.
The backwards compatibility part of the PR looks ok.

**Review by plkokanov — COMMENTED** [review]
I just have one concern, although I think the chances of it happening are quite slim - if a shoot cluster has not been reconciled for a long time and we try to migrate it before it gets reconciled, but after this change is introduced, the data from the old `etcd-encryption-secret` will not be restored. In that case the `etcdEncryptionConfiguration ` will still be present in the shoot state, but it will not get deployed on the new seed since from what I can see in `InitializeSecretsManagement` only secrets that have `type==secret` and the `managed-by: secrets-manager` label are deployed snd the steps that deal with the `etcdEncryptionConfiguration ` have been removed.
There's a step in the migration flow that deploys the kube-apiserver (`WakeUpKubeAPIServer`) and therefore would also do the necessary steps to replace the old with the new encryption secrets, but currently it is only executed if the shoot is hibernated and the kube-apiserver deployment still exists.

Otherwise I think it lgtm.

**Comment by rfranzke** [comment]
Yes, there are a few more of such corner cases regarding migration, I believe. What do you suggest? Given that the migration feature is still alpha, I would not like to invest too much into it, TBH.

**Comment by rfranzke** [comment]
I could offer to extend your validation introduced in #5587 with an additional check to only allow migration of the new encryption secret is found in the `ShootState`.

**Comment by plkokanov** [comment]
>I could offer to extend your validation introduced in https://github.com/gardener/gardener/pull/5587 with an additional check to only allow migration of the new encryption secret is found in the ShootState.

That is probably better than what I was about to suggest - to keep the `etcdEncryptionConfiguration` data in the `ShootState` for now and delete it in the next release. Then if something goes wrong we can restore the data manually.

**Comment by rfranzke** [comment]
OK, then let's merge your PR, and I will rebase on top and add the check. Thanks!

**Review by plkokanov — APPROVED** [review]
/lgtm

**Pull request merged**